asdfghjkl/precondition.py

```diff
@@ -66,15 +66,15 @@ def update_curvature(self, inputs=None, targets=None, data_loader=None):
                                        inputs=inputs,
                                        targets=targets,
                                        data_loader=data_loader,
-                                       fisher_types=self.fisher_type,
+                                       fisher_type=self.fisher_type,
                                        fisher_shapes=self.fisher_shape,
                                        n_mc_samples=self.n_mc_samples)
         self.fisher_manager = rst
 
     def move_curvature(self, postfix, scale=1., to_pre_inv=False):
         self.accumulate_curvature(postfix, scale, to_pre_inv, replace=True)
 
-    def accumulate_curvature(self, postfix='acc', scale=1., to_pre_inv=False, replace=False):
+    def accumulate_curvature(self, postfix='acc', scale=1., smoothing_weight=None, to_pre_inv=False, replace=False):
```

Comment:
@Cecilwang how about `ema_decay` (ema: exponential moving average) instead of `smoothing_weight`?
The former is more descriptive.

Reply:
https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average

Reply:
Ok, but one more question: which formula do we prefer?
1. ema_value = ema_value * ema_decay + new_value * (1 - ema_decay)  // from tensorflow
2. ema_value = ema_value * (1 - ema_decay) + new_value * ema_decay  // from wikipedia

Reply:
formula 2.
because an EMA is often defined as
ema_value = ema_value + ema_decay * (new_value - ema_value)
as is in the wiki page.